src/Framework/TestCase.php

```diff
@@ -1645,6 +1688,10 @@ protected function getObjectForTrait($traitName, array $arguments = [], $traitCl
      * @throws \Prophecy\Exception\Doubler\ClassNotFoundException
      * @throws \Prophecy\Exception\Doubler\DoubleException
      * @throws \Prophecy\Exception\Doubler\InterfaceNotFoundException
+     *
+     * @psalm-template RealInstanceType of object
+     * @psalm-param class-string<RealInstanceType>|null $classOrInterface
+     * @psalm-return ObjectProphecy<RealInstanceType>
```

Comment:
To be useful I suppose this needs to have the @template tag annotation added to https://github.com/phpspec/prophecy/blob/7e272180527c34a97680de85eb5aba0847a664e0/src/Prophecy/Prophecy/ObjectProphecy.php#L30 (or a stub for it)